otelcol/collector.go

```diff
@@ -116,6 +116,9 @@ func NewCollector(set CollectorSettings) (*Collector, error) {
 	var err error
 	configProvider := set.ConfigProvider
 
+	set.ConfigProviderSettings.ResolverSettings.ProviderSettings = confmap.ProviderSettings{Logger: zap.NewNop()}
```

Comment:
I wanted to avoid a dependency on `confmaptest` so I just directly instantiated a no-op logger here in preparation for actually instantiating a logger. I'm open to alternate approaches if we want to encapsulate this somehow.

Reply:
I updated `confmap.NewResolver` to use a no-op logger if no logger is provided, so this code can just be updated to pass in the actual logger once that is implemented.